service/src/test/java/bio/terra/cbas/util/methods/TestGithubUrlDetailsManager.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.cbas.util.methods;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.net.URISyntaxException;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.test.context.ContextConfiguration;
+
+@WebMvcTest
+@ContextConfiguration(classes = {GithubUrlDetailsManager.class})
+class TestGithubUrlDetailsManager {
+
+  @Test
+  void returnsExpectedUrlComponents() throws URISyntaxException {
```

Comment:
This would be a great time to use table based testing. You could define a big table of input and expected org / repo / branch / etc names, and then validate them all with the same test method. And you could also list a bunch of failure cases and check we handle those nicely too